hypervisor/dm/io_req.c

```diff
@@ -616,11 +610,60 @@ void register_pio_emulation_handler(struct acrn_vm *vm, uint32_t pio_idx,
 	vm->emul_pio[pio_idx].io_write = io_write_fn_ptr;
 }
 
+/**
+ * @brief Find match MMIO node
+ *
+ * This API find match MMIO node from \p vm.
+ *
+ * @param vm The VM to which the MMIO node is belong to.
+ *
+ * @return If there's a match mmio_node return it, otherwise return NULL;
+ */
+static inline struct mem_io_node *find_match_mmio_node(struct acrn_vm *vm,
+				uint64_t start, uint64_t end, bool update_idx)
+{
+	bool found = false;
+	uint16_t idx;
+	struct mem_io_node *mmio_node;
+
+	for (idx = 0U; idx < CONFIG_MAX_EMULATED_MMIO_REGIONS; idx++) {
+		mmio_node = &(vm->emul_mmio[idx]);
+		if ((mmio_node->range_start == start) && (mmio_node->range_end == end)) {
+			found = true;
+			if (update_idx && (idx > vm->max_emul_mmio_regions)) {
+				vm->max_emul_mmio_regions = idx;
+			}
```

Comment:
update is better to be done in caller side.

This API is "find_xxx"  ... Normally Find will never change internal status.

Reply:
@dongyaozu  Yes. This's refined.